js/apiml.js

```diff
@@ -0,0 +1,194 @@
+const Promise = require("bluebird");
+const eureka = require('eureka-js-client').Eureka;
+const zluxUtil = require('./util');
+
+const log = zluxUtil.loggers.apiml;
+
+const MEDIATION_LAYER_EUREKA_DEFAULTS = {
+  "preferSameZone": false,
+  "requestRetryDelay": 10000,
+  "heartbeatInterval": 3000,
+  "registryFetchInterval": 10000,
+  "fetchRegistry": false,
+  "availabilityZones": {
+    "defaultZone": ["defaultZone"]
+  }, 
+};
+
+
+const MEDIATION_LAYER_INSTANCE_DEFAULTS = {
+  instanceId: "localhost:zowe-zlux:8543",
+  app: "zlux",
+  hostName: "localhost",
+  ipAddr: "127.0.0.1", 
+  vipAddress: "localhost",
+  status: "UP",
+  dataCenterInfo: {
+    '@class': 'com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo',
+    name: 'MyOwn'
+  },
+  leaseInfo: {
+    durationInSecs: 10,
+    renewalIntervalInSecs: 10
+  },
+  metadata: {
+    "routed-services.1.gateway-url": "/api/v1",
+    "routed-services.1.service-url": "/ZLUX",
+    "routed-services.2.gateway-url": "/ui/v1",
+    "routed-services.2.service-url": "/ZLUX",
+    "routed-services.3.gateway-url": "/ws/v1",
+    "routed-services.3.service-url": "/ZLUX",
+    "routed-services.4.gateway-url": "/v1/api-doc",
+    "routed-services.4.service-url": "/ZLUX",
+
+    'mfaas.discovery.catalogUiTile.id': 'zlux',
+    'mfaas.discovery.catalogUiTile.title': 'ZLux Proxy Server',
```

Comment:
We really need a name change. @jhenryRocket don't we use "Zowe Application Server"? I'd also find "zlux server" to be appropriate. But, I think we need to get "proxy" out of the name because when people see Zowe, they see components and zlux's server's specialty is more about serving apps than proxying, even though it can do both.

Reply:
Yes - we use "Zowe Application Server" in the user doc. I agree that we should remove the "proxy" part.